tools/c7n_azure/c7n_azure/resources/event_hub.py

```diff
@@ -46,3 +54,29 @@ class resource_type(ArmResourceManager.resource_type):
             'properties.isAutoInflateEnabled'
         )
         resource_type = 'Microsoft.EventHub/namespaces'
+
+
+@EventHub.filter_registry.register('firewall-rules')
+class EventHubFirewallRulesFilter(FirewallRulesFilter):
+
+    def __init__(self, data, manager=None):
+        super(EventHubFirewallRulesFilter, self).__init__(data, manager)
+        self._log = logging.getLogger('custodian.azure.eventhub')
+        self.client = None
+
+    @property
+    def log(self):
+        return self._log
+
+    def process(self, resources, event=None):
+        self.client = self.manager.get_client()
```

Comment:
nit: wondering if we should move this code `self.client = self.manager.get_client()` to the `FirewallRulesFilter` class, we do pass manager either way..

Reply:
I think needing the client is specific only to this subclass though?  Most of the other resources have nothing but a dictionary lookup in the subclass - this one does an API call.